url7d636e8121_laforge49_JActor2_tgz-pJ8-org_agilewiki_jactor2_core_revisited_EHJ8/src/org/agilewiki/jactor2/core/impl/Plant.java

```diff
@@ -0,0 +1,40 @@
+package org.agilewiki.jactor2.core.impl;
+
+import org.agilewiki.jactor2.core.impl.mtPlant.PlantConfiguration;
+import org.agilewiki.jactor2.core.impl.mtPlant.PlantMtImpl;
+import org.agilewiki.jactor2.core.plant.impl.PlantBase;
+
+/**
+ * Plant is a singleton and is the top-level object.
+ * The plant has an internal reactor which is the root of a tree of all reactors,
+ * though this tree is made using weak references.
+ */
+final public class Plant extends PlantBase {
+
+    /**
+     * Create a plant with the default configuration.
+     */
+    public Plant() throws Exception {
+        new PlantMtImpl();
+    }
+
+    /**
+     * Create a plant with the default configuration,
+     * but with the given reactor thread pool size.
+     *
+     * @param _reactorThreadPoolSize The number of threads to be created for the
+     *                               reactor thread pool.
+     */
+    public Plant(final int _reactorThreadPoolSize) throws Exception {
+        new PlantMtImpl(_reactorThreadPoolSize);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the resource returned by the *PlantMtImpl* constructor in a try-finally block. Execution paths that do not contain closure statements were detected.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).